com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -1172,7 +1232,8 @@ void AgentStep()
             if ((m_RequestAction) && (m_Brain != null))
             {
                 m_RequestAction = false;
-                OnActionReceived(m_Action.vectorActions);
+                OnActionReceived(vectorActuator.Actions);
+                actuators.ExecuteActions();
```

Comment:
I need to think about this - pretty sure you'll call the vectorActuator's OnActionReceived() too, but maybe that's a no-op.

Another option (which I think we considered for sensors but didn't do) would be something like
```
class AgentActuator : IActuator {
  Agent m_Agent;
  void OnActionReceived()
  {
    m_Agent.OnActionReceived(this.Actions);
  } 
}
```
Then you don't need to do anything special here to trigger Agent.OnActionReceived()

Reply:
And if you go with my ActuatorSpace struct idea, you can probably get rid of ContinuousVectorActuator and DiscreteVectorActuator, and just return a (cached) ActuatorSpace based on the Agent's behavior parameters.

Reply:
I thought about having the Actuator reference the Agent, but then we get into a cyclic dependency, It would be nice if Agent had an interface we could interact with.